File: python/Mahasiswa.py

# Saya [Elsa Nabiilah] [2108805]
# mengerjakan LATIHAN 2 dalam mata kuliah Desain Pemrograman Berorientasi Objek
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
# Aamiin.

from Human import Human


class Mahasiswa(Human):

    # class mahasiswa mejadi child/toodler dari class human karena
    # atribut Mahasiswa melekat kaitannya dengan atribut Human karena sam- sama mengarah
    # kepada point-point identitas
    # deklarasi private atribut

    def __init__(self):
        self.__nim = ""
        self.__fak = ""
        self.__prodi = ""
    #  self

    # setter getter method
    def getnim(self):
        # getter nim
        return self.__nim

    def setnim(self, nim):
        # setter nim
        self.__nim = nim

    def getfak(self):
        # getter fakultas
        return self.__fak

    def setfak(self, fak):
        # setter fakultas
        self.__fak = fak

    def getprof(self):
        # getter prodi
        return self.__prodi

    def setprodi(self, prodi):
        # setter prodi
        self.__prodi = prodi

    # function untuk menampilkan per atribut
    def printnim(self):
        print("NIM           : ", self.__nim)

    def printps(self):
        print("Program Studi : ", self.__prodi)

    def printfak(self):
        print("Fakultas      : ", self.__fak)

File: php/Mahasiswa.php
<?php
/*
Saya [Elsa Nabiilah] [2108805]
mengerjakan LATIHAN 2 dalam mata kuliah Desain Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/
include "Human.php";

class Mahasiswa extends Human
{
    // class mahasiswa mejadi child/toodler dari class human karena
    // atribut Mahasiswa melekat kaitannya dengan atribut Human karena sam- sama mengarah
    // kepada point-point identitas
    // deklarasi private atribut
    private $nim;
    private $fak;
    private $prodi;

    function __construct() //($nim, $fak, $prodi)
    {
        // $this->nim = $nim;
        // $this->fak = $fak;
        // $this->prodi = $prodi;
    }
    // setter getter method
    public function getnim()
    {
        // getter nim
        return $this->nim;
    }

    public function setnim($nim)
    {
        // setter nim
        $this->nim = $nim;
    }

    public function getfak()
    {
        // getter fakultas
        return $this->fak;
    }

    public function setfak($fak)
    { // setter fakultas
        $this->fak = $fak;
    }

    public function getprof()
    {
        // getter prodi
        return $this->prodi;
    }

    public function setprodi($prodi)
    { // setter prodi
        $this->prodi = $prodi;
    }

    // function untuk menampilkan per atribut
    public function printnim()
    {
        // echo "<tr><td>";
        echo "NIM           : " . $this->getnim() . "<br/>";
        //echo "</td><td>";
        //echo "Email Univ    : " . $this->getemail() . "<br/>";
    }
    public function printps()
    {
        // echo "<tr><td>";
        echo "Program Studi : " . $this->getprof() . "<br/>";
        //echo "</td><td>";
    }
    public function printfak()
    {
        echo "Fakultas      : " . $this->getfak() . "<br/>";
    }

    function __destruct()
    {
    }
}
